File: SDKs/linode.com/4.97.2/python/src/sdk/models/shared/diskrequest.py
from dataclasses import dataclass, field
from typing import Any,List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class DiskRequest:
    r"""DiskRequest
    Disk object request.
    """
    
    label: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('label') }})
    size: int = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('size') }})
    authorized_keys: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('authorized_keys') }})
    authorized_users: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('authorized_users') }})
    filesystem: Optional[FilesystemEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filesystem') }})
    image: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('image') }})
    root_pass: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('root_pass') }})
    stackscript_data: Optional[dict[str, Any]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('stackscript_data') }})
    stackscript_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('stackscript_id') }})
    

File: SDKs/linode.com/4.97.2/typescript/src/sdk/models/shared/diskrequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { FilesystemEnum } from "./filesystemenum";



// DiskRequest
/** 
 * Disk object request.
**/
export class DiskRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=authorized_keys" })
  authorizedKeys?: string[];

  @SpeakeasyMetadata({ data: "json, name=authorized_users" })
  authorizedUsers?: string[];

  @SpeakeasyMetadata({ data: "json, name=filesystem" })
  filesystem?: FilesystemEnum;

  @SpeakeasyMetadata({ data: "json, name=image" })
  image?: string;

  @SpeakeasyMetadata({ data: "json, name=label" })
  label: string;

  @SpeakeasyMetadata({ data: "json, name=root_pass" })
  rootPass?: string;

  @SpeakeasyMetadata({ data: "json, name=size" })
  size: number;

  @SpeakeasyMetadata({ data: "json, name=stackscript_data" })
  stackscriptData?: Map<string, any>;

  @SpeakeasyMetadata({ data: "json, name=stackscript_id" })
  stackscriptId?: number;
}
